src/docker-osx-dev

```diff
@@ -235,8 +234,9 @@ function do_rsync {
 #
 function sync {
   local readonly paths_to_sync=("$@")
-
-  for path in "${paths_to_sync[@]}"; do
+  for path in ${paths_to_sync[@]}
```

Comment:
It turns out you're on the right track, but a better fix is to remove the quotes from the line above. I just did that in another commit, so I'll close this PR. Thanks for helping track this down!